cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/converter/MetricConverter.java

```diff
@@ -0,0 +1,43 @@
+package com.sap.cloud.cf.monitoring.java.converter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.sap.cloud.cf.monitoring.client.model.Metric;
+
+public abstract class MetricConverter<T> {
+    private static final String KEY_TYPE = "type";
+
+    public enum MetricType {
+                     TIMER("timer"), HISTOGRAM("histogram"), GAUGE("gauge"), METER("meter"), COUNTER("counter");
+
+        private final String metricTypeName;
+
+        String getMetricTypeName() {
+            return metricTypeName;
+        }
+
+        private MetricType(String metricTypeName) {
+            this.metricTypeName = metricTypeName;
+        }
+    }
+
+    public List<Metric> convert(Map<String, T> metrics, long timestamp) {
+        ArrayList<Metric> result = new ArrayList<>();

```

Comment:
At other places Java 8 Streams were used. Since `convertMetricEntry` does not declare an exception, you could use this here as well.

Reply:
Done. During the code review I changed the java version from 7 to 8. That is the reason for skipping streams here.